Sources/CocoaCollectionViewLayouts/CollectionViewGridLayout/CollectionViewGridLayout.swift

```diff
@@ -273,12 +273,9 @@ public class CollectionViewGridLayout: NSCollectionViewLayout {
         }
     }
     
-    private var _oldBounds: NSRect = .zero
-    
     public override func shouldInvalidateLayout(forBoundsChange newBounds: NSRect) -> Bool {
         // invalidate layout only when the visible width changed
-        let invalidate = scrollDirection == .vertical ? newBounds.width != _oldBounds.width : newBounds.height != _oldBounds.height
-        _oldBounds = newBounds
+        let invalidate = scrollDirection == .vertical ? newBounds.width != _contentSize.width : false
```

Comment:
Returning `false` in else case would refuse relayout operations for bounds change in horizontally scrolling layout.

Replace `false` with 

```swift
newBounds.height != _contentSize.height
```

can be an easy solution.

Also, since there's no need to cache the value of `invalidate`, we can remove the `invalidate` and return the result directly.